src/tests/ftest/util/host_utils.py

```diff
@@ -0,0 +1,338 @@
+        self._clients = HostRole()
+        self.access_points = NodeSet()
+
+    @property
+    def all(self):
+        """Get all of the server and client hosts.
+
+        Returns:
+            NodeSet: all the server and client hosts
+
+        """
+        return self.servers | self.clients
+
+    @property
+    def servers(self):
+        """Get the server hosts.
+
+        Returns:
+            NodeSet: the server hosts
+        """
+        return self._servers.hosts
+
+    @property
+    def clients(self):
+        """Get the clients hosts.
+
+        Returns:
+            NodeSet: the clients hosts
+        """
+        return self._clients.hosts
+
+    @property
+    def server_partition(self):
+        """Get the server partition information.
+
+        Returns:
+            HostPartition: the server partition
+        """
+        return self._servers.partition
+
+    @property
+    def client_partition(self):
+        """Get the client partition information.
+
+        Returns:
+            HostPartition: the client partition
+        """
+        return self._clients.partition
+
+    @property
+    def clients_with_localhost(self):
+        """Get the test clients including the localhost.
+
+        Returns:
+            NodeSet: test clients including the localhost
+
+        """
+        return self.clients | NodeSet(get_local_host())
+
+    def display(self, log):
+        """Display the host information.
+
+        Args:
+            log (logger): logger for the messages produced by this method
+        """
+        log.info("-" * 100)
+        log.info("--- HOST INFORMATION ---")
+        log.info("hostlist_servers:    %s", self.servers)
+        log.info("hostlist_clients:    %s", self.clients)
+        log.info("server_partition:    %s", self._servers.partition.name)
+        log.info("client_partition:    %s", self._clients.partition.name)
+        log.info("server_reservation:  %s", self._servers.partition.reservation)
+        log.info("client_reservation:  %s", self._clients.partition.reservation)
+        log.info("access_points:       %s", self.access_points)
+
+    def set_hosts(self, log, server_hosts, server_partition, server_reservation, client_hosts,
+                  client_partition, client_reservation):
+        """Set the hosts.
+
+        Args:
+            log (logger): logger for the messages produced by this method
+            server_hosts (object): hosts to define as servers
+            server_partition (str): server partition name
+            server_reservation (str): server reservation name
+            client_hosts (object): hosts to define as clients
+            client_partition (str): client partition name
+            client_reservation (str): client reservation name
+
+        Raises:
+            HostException: if there is a problem obtaining the hosts
+
+        """
+        self._servers = HostRole(server_hosts, server_partition, server_reservation)
+        self._servers.update_hosts(log)
```

Comment:
Why doesn't `HostRole.__init__` handle the update? I think that could eliminate a lot of the set and update functions which don't take any arguments or return anything